core/jvm/src/test/scala/fs2/CompressSpec.scala

```diff
@@ -168,5 +513,454 @@ class CompressSpec extends Fs2Spec {
+              .through(compress.gzip(bufferSize = chunkSize))
+              .through(classic.gunzip(bufferSize = chunkSize))
+          )).compile.toVector
+          .asserting { vector =>
+            println(s"gzip |> gunzip duration with random bytes (compress, classic): ${vector
+              .map { case (_, duration) => duration }}")
+            vector.size shouldBe 2
+            vector.head._1 shouldBe bytesIn
+            vector(1)._1 shouldBe bytesIn
+          }
+      }
+
+      "with zero bytes" in {
+        val bytesIn: Long = 64 * 1024 * 1024
+        val chunkSize = 32 * 1024
+        (toLengthAndDurationMs(
+          constantByteStream[IO](bytesIn, chunkSize, 0)
+            .through(compress.gzip(bufferSize = chunkSize))
+            .through(compress.gunzip(bufferSize = chunkSize))
+        ) ++
+          toLengthAndDurationMs(
+            constantByteStream[IO](bytesIn, chunkSize, 0)
+              .through(compress.gzip(bufferSize = chunkSize))
+              .through(classic.gunzip(bufferSize = chunkSize))
+          )).compile.toVector
+          .asserting { vector =>
+            println(s"gzip |> gunzip duration with zero bytes (compress, classic): ${vector
+              .map { case (_, duration) => duration }}")
+            vector.size shouldBe 2
+            vector.head._1 shouldBe bytesIn
+            vector(1)._1 shouldBe bytesIn
+          }
+      }
+
+    }
+
+    def toEncodableFileName(fileName: String): String =
+      new String(
+        fileName.replaceAll("\u0000", "_").getBytes(StandardCharsets.ISO_8859_1),
+        StandardCharsets.ISO_8859_1
+      )
+
+    def toEncodableComment(comment: String): String =
+      new String(
+        comment.replaceAll("\u0000", " ").getBytes(StandardCharsets.ISO_8859_1),
+        StandardCharsets.ISO_8859_1
+      )
+
+    def randomByteStream[F[_]](bytes: Long, chunkSize: Int, random: Random): Stream[F, Byte] = {
+      def go(bytesRemaining: Long): Pull[F, Byte, Unit] = {
+        val nextBytesRemaining = bytesRemaining - chunkSize
+        if (nextBytesRemaining > 0) {
+          val buffer = Array.ofDim[Byte](chunkSize)
+          random.nextBytes(buffer)
+          Pull.output(Chunk.bytes(buffer)) >> go(nextBytesRemaining)
+        } else {
+          val buffer = Array.ofDim[Byte](bytesRemaining.toInt)
+          random.nextBytes(buffer)
+          Pull.output(Chunk.bytes(buffer))
+        }
+      }
+      go(bytes).stream
+    }
+
+    def constantByteStream[F[_]](bytes: Long, chunkSize: Int, byte: Byte): Stream[F, Byte] = {
+      def go(bytesRemaining: Long): Pull[F, Byte, Unit] = {
+        val nextBytesRemaining = bytesRemaining - chunkSize
+        if (nextBytesRemaining > 0)
+          Pull.output(Chunk.bytes(Array.fill(chunkSize)(byte))) >> go(nextBytesRemaining)
+        else Pull.output(Chunk.bytes(Array.fill(bytesRemaining.toInt)(byte)))
+      }
+      go(bytes).stream
+    }
+
+  }
+}
+
+object classic {
```

Comment:
This seems unnecessary. The tests as they are written without `classic` are sufficient coverage. It would be *nice* to have an optional test which delegates to the `gzip` command if it is present on the `PATH` (we did this for fs2-gzip originally, but representing it in ScalaTest is harder than in Specs2).

Reply:
Agreed, this can be removed. Retaining the original implementation helped sanity check compatibility.

Integrating with the system's gzip/gunzip if on the path is a nice idea and much more powerful that the one test I added to decompress the rudimentary unix.gzip file I created and copied as hex bytes into the tests.

Reply:
`classic` implementation no longer required in the tests as it needs to remain remain to be deprecated for the necessary type changes to be implemented in fs2.compress2. (Opinions on naming are welcome.)